lib/cext/ruby/io.h

```diff
@@ -85,7 +85,7 @@ typedef struct rb_io_t {
 /* #define FMODE_INET                  0x00400000 */
 /* #define FMODE_INET6                 0x00800000 */
 
-#define GetOpenFile(file, pointer) ( (pointer) = truffle_managed_malloc(sizeof(rb_io_t)), (pointer)->mode = FIX2INT(rb_iv_get(file, "@mode")), (pointer)->fd = FIX2INT(rb_iv_get(file, "@descriptor")), rb_io_check_closed(pointer) )
+#define GetOpenFile(file, pointer) ( (pointer) = truffle_invoke(RUBY_CEXT, "GetOpenFile", file), rb_io_check_closed(pointer) )
```

Comment:
Now that we have a single call to create it, we could do:
```c
rb_io_check_closed((pointer) = truffle_invoke(RUBY_CEXT, "GetOpenFile", file))
```
which is more similar to https://github.com/ruby/ruby/blob/v2_3_3/include/ruby/io.h#L120

Reply:
Right... Comma operator... I misinterpreted what this does.
C is such a nice language ;)